distributed/worker.py

```diff
@@ -1930,6 +1935,8 @@ def transition_missing_fetch(
         ts.state = "fetch"
         ts.done = False
         self.data_needed.push(ts)
+        for w in ts.who_has:
+            self.data_needed_per_worker[w].push(ts)
```

Comment:
Fixes bug where tasks transitioning from fetch to missing would not be picked up by ``select_keys_for_gather``